base/replutil.jl

```diff
@@ -205,7 +205,11 @@ function showerror(io::IO, ex::MethodError)
         print(io, "This may have arisen from a call to the constructor $construct_type(...),",
                   "\nsince type constructors fall back to convert methods.")
     end
-    show_method_candidates(io, ex)
+    try
+        show_method_candidates(io, ex)
+    catch
+        warn("Unable to show method candidates")
```

Comment:
The rest of the output is going to io, seems odd to have part of it not. What about in IJulia, does this result in the warning going to the terminal that's running the kernel instead of the output cell?